dart/test/sentry_event_test.dart

```diff
@@ -309,15 +309,41 @@ void main() {
       expect(serialized['threads']['values'], isNotNull);
     });
 
+    test('should serialize $SentryThread when id matches exception id', () {
+      final serialized = SentryEvent(
+        exceptions: [
+          SentryException(
+            type: 'foo',
+            value: 'bar',
+            threadId: 0,
+          )
+        ],
+        threads: [
+          SentryThread(
+            id: 0,
+            crashed: true,
+            current: true,
+            name: 'Isolate',
+          )
+        ],
+      ).toJson();
+      expect(serialized['threads']?['values'], isNotEmpty);
+    });
+
     test(
```

Comment:
I think this test was also wrong before. After adding some expectations, this one also fails.